app/src/main/java/de/tum/in/tumcampusapp/fragments/PdfViewFragment.java

```diff
@@ -37,7 +40,32 @@ public void onResume() {
         pdfView.fromFile(pdf).load();
     }
 
-    public void setPdf(File pdf){
-        this.pdf = pdf;
+    public boolean setPdf(File pdf){
+        if (isFilePdf(pdf)){
+            this.pdf = pdf;
+            return true;
+        }
+        return false;
+    }
+
+    private boolean isFilePdf(File file){
```

Comment:
E.g. if we receive a 404 from the mvv website, it is still downloaded as html and therefore it's necessary to check if we're trying to open a valid pdf (first bytes starting with %PDF). From my experiments the PdfView's implementation would just be blank, if it tries to open a non-pdf file.
Therefore I added this method as an additional check before opening a file. If you prefer that I redesign the code to use a custom "FileIsNotPdfException", I can do so. In my opinion our only purpose of checking whether a file is a valid pdf or not, is to open it. And because this responsibility only lies in the PdfViewerFragment, I think it's in a valid spot there. Let me know if that fits your idea.